lib/natalie/compiler2/pass1.rb

```diff
@@ -386,13 +386,12 @@ def transform_yield(exp, used:)
       # returns a pair of [name, prep_instruction]
       # prep_instruction being the instruction(s) needed to get the owner of the constant
       def constant_name(name)
-        if name.is_a?(Symbol)
+        case name
+        in Symbol
           [name, PushSelfInstruction.new]
-        elsif name.sexp_type == :colon2
-          _, namespace, name = name
+        in [:colon2, namespace, name]
           [name, transform_expression(namespace, used: true)]
-        elsif name.sexp_type == :colon3
-          _, name = name
+        in [:colon3, name]
```

Comment:
I'm not sure if I'm ready to leave Ruby 2.7 behind since it's so ubiquitous on Linux distros... is there a way to disable the pattern matching warning on Ruby 2.7?

>warning: Pattern matching is experimental, and the behavior may change in future versions of Ruby!

Reply:
@FnControlOption any thoughts on this...?